cmd/sealer/cmd/utils/cluster_test.go

```diff
@@ -15,54 +15,75 @@
 package utils
 
 import (
-	"fmt"
 	"net"
 	"testing"
 
 	"github.com/sealerio/sealer/cmd/sealer/cmd/types"
 	"github.com/sealerio/sealer/pkg/clusterfile"
-	"github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/assert"
 )
 
+//func Test_returnFilteredIPList1(t *testing.T) {
+//	tests := []struct {
+//		name              string
+//		clusterIPList     []net.IP
+//		toBeDeletedIPList []net.IP
+//		IPListExpected    []net.IP
+//	}{
+//		{
+//			name:              "test return Filtered IP List",
+//			clusterIPList:     []net.IP{net.ParseIP("10.10.10.1"), net.ParseIP("10.10.10.2"), net.ParseIP("10.10.10.3"), net.ParseIP("10.10.10.4")},
+//			toBeDeletedIPList: []net.IP{net.ParseIP("10.10.10.1"), net.ParseIP("10.10.10.2"), net.ParseIP("10.10.10.3"), net.ParseIP("10.10.10.4")},
+//		},
+//	}
+//}
+
 func Test_returnFilteredIPList(t *testing.T) {
 	tests := []struct {
 		name              string
 		clusterIPList     []net.IP
 		toBeDeletedIPList []net.IP
+		IPListExpected    []net.IP
 		wantErr           bool
 	}{
 		{
 			"test",
 			[]net.IP{net.ParseIP("10.10.10.1"), net.ParseIP("10.10.10.2"), net.ParseIP("10.10.10.3"), net.ParseIP("10.10.10.4")},
 			[]net.IP{net.ParseIP("10.10.10.1"), net.ParseIP("10.10.10.2"), net.ParseIP("10.10.10.3"), net.ParseIP("10.10.10.4")},
+			[]net.IP{},
 			false,
 		},
 		{
 			"test1",
 			[]net.IP{net.ParseIP("10.10.10.1"), net.ParseIP("10.10.10.2"), net.ParseIP("10.10.10.3"), net.ParseIP("10.10.10.4")},
 			[]net.IP{},
+			[]net.IP{net.ParseIP("10.10.10.1"), net.ParseIP("10.10.10.2"), net.ParseIP("10.10.10.3"), net.ParseIP("10.10.10.4")},
 			false,
 		},
 		{
 			"test2",
 			[]net.IP{net.ParseIP("10.10.10.1"), net.ParseIP("10.10.10.2"), net.ParseIP("10.10.10.3"), net.ParseIP("10.10.10.4")},
 			[]net.IP{net.ParseIP("10.10.10.4")},
+			[]net.IP{net.ParseIP("10.10.10.1"), net.ParseIP("10.10.10.2"), net.ParseIP("10.10.10.3")},
 			false,
 		},
 		{
 			"test3",
 			[]net.IP{},
 			[]net.IP{net.ParseIP("10.10.10.1"), net.ParseIP("10.10.10.2"), net.ParseIP("10.10.10.3"), net.ParseIP("10.10.10.4")},
+			[]net.IP{},
 			false,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			if res := removeIPList(tt.clusterIPList, tt.toBeDeletedIPList); (res != nil) != tt.wantErr {
-				fmt.Println(res)
+				assert.NotNil(t, res)
```

Comment:
"(res != nil) != tt.wantErr"
"assert.NotNil(t, res)"
Are them duplicated?
IMO, both of them are unnecessary.
Just do "assert.Equal(t, tt.IPListExpected, res)".
In addition, I guess "if !result " will not be executed if "assert" error happened?

Reply:
Yes, the judgment does repeat itself